commit 808e6e410119d25161ad530072a35886d81c18a4
Author: Susan Franklin <sfranklin>
Date:   Fri Oct 26 16:16:34 2007 +0000

    Bug 207513 - [prov] Admin UI repo views don't update on delete

0	4	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ArtifactRepositoriesView.java
0	4	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/MetadataRepositoriesView.java
1	2	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
3	1	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
1	2	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
2	4	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/operations/ProvisioningUtil.java
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ArtifactRepositoriesView.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ArtifactRepositoriesView.java
index a8268b0..dcdec8a 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ArtifactRepositoriesView.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ArtifactRepositoriesView.java
@@ -50,10 +50,6 @@ public class ArtifactRepositoriesView extends RepositoriesView {
 		return ProvAdminUIMessages.ArtifactRepositoriesView_AddRepositoryTooltip;
 	}
 
-	protected String getRemoveCommandLabel() {
-		return ProvAdminUIMessages.ArtifactRepositoriesView_RemoveRepositoryLabel;
-	}
-
 	protected String getRemoveCommandTooltip() {
 		return ProvAdminUIMessages.ArtifactRepositoriesView_RemoveRepositoryTooltip;
 	}
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/MetadataRepositoriesView.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/MetadataRepositoriesView.java
index eb1ee50..4faf01e 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/MetadataRepositoriesView.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/MetadataRepositoriesView.java
@@ -66,10 +66,6 @@ public class MetadataRepositoriesView extends RepositoriesView {
 		return ProvAdminUIMessages.MetadataRepositoriesView_AddRepositoryTooltip;
 	}
 
-	protected String getRemoveCommandLabel() {
-		return ProvAdminUIMessages.MetadataRepositoriesView_RemoveRepositoryLabel;
-	}
-
 	protected String getRemoveCommandTooltip() {
 		return ProvAdminUIMessages.MetadataRepositoriesView_RemoveRepositoryTooltip;
 	}
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
index 33b4252..8999f42 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
@@ -32,18 +32,17 @@ public class ProvAdminUIMessages extends NLS {
 	public static String MetadataRepositoriesView_AddRepositoryTooltip;
 	public static String MetadataRepositoriesView_AddRepositoryLabel;
 	public static String MetadataRepositoriesView_ChooseProfileDialogTitle;
-	public static String MetadataRepositoriesView_RemoveRepositoryLabel;
 	public static String MetadataRepositoriesView_RemoveRepositoryTooltip;
 	public static String MetadataRepositoriesView_RemoveRepositoryOperationLabel;
 	public static String ArtifactRepositoriesView_AddRepositoryTooltip;
 	public static String ArtifactRepositoriesView_AddRepositoryLabel;
-	public static String ArtifactRepositoriesView_RemoveRepositoryLabel;
 	public static String ArtifactRepositoriesView_RemoveRepositoryTooltip;
 	public static String ArtifactRepositoriesView_RemoveRepositoryOperationLabel;
 	public static String ProfilesView_AddProfileTooltip;
 	public static String ProfilesView_AddProfileLabel;
 	public static String ProfilesView_RemoveProfileLabel;
 	public static String ProfilesView_RemoveProfileTooltip;
+	public static String RepositoriesView_RemoveCommandLabel;
 	// Preferences
 	public static String InstalledIUPropertyPage_NoInfoAvailable;
 	public static String ProvisioningPrefPage_HideInternalRepos;
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
index 1fd66cf..42783b2 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/RepositoriesView.java
@@ -152,7 +152,9 @@ abstract class RepositoriesView extends ProvView {
 
 	protected abstract String getAddCommandTooltip();
 
-	protected abstract String getRemoveCommandLabel();
+	protected String getRemoveCommandLabel() {
+		return ProvAdminUIMessages.RepositoriesView_RemoveCommandLabel;
+	}
 
 	protected abstract String getRemoveCommandTooltip();
 
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
index 2224627..2066730 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
@@ -20,13 +20,12 @@ AddMetadataRepositoryDialog_OperationLabel=Add Metadata Repository
 AddArtifactRepositoryDialog_OperationLabel=Add Artifact Repository
 MetadataRepositoriesView_AddRepositoryLabel=Add repository...
 MetadataRepositoriesView_AddRepositoryTooltip=Add a new metadata repository
-MetadataRepositoriesView_RemoveRepositoryLabel=Remove
+RepositoriesView_RemoveCommandLabel=Remove
 MetadataRepositoriesView_RemoveRepositoryTooltip=Remove the selected metadata repositories
 MetadataRepositoriesView_ChooseProfileDialogTitle=Choose Profile
 MetadataRepositoriesView_RemoveRepositoryOperationLabel=Remove Metadata Repositories
 ArtifactRepositoriesView_AddRepositoryLabel=Add repository...
 ArtifactRepositoriesView_AddRepositoryTooltip=Add a new artifact repository
-ArtifactRepositoriesView_RemoveRepositoryLabel=Remove repositories
 ArtifactRepositoriesView_RemoveRepositoryTooltip=Remove the selected artifact repositories
 ArtifactRepositoriesView_RemoveRepositoryOperationLabel=Remove Artifact Repositories
 ProfilesView_AddProfileTooltip=Add a new profile
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/operations/ProvisioningUtil.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/operations/ProvisioningUtil.java
index f99743b..cdc24f9 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/operations/ProvisioningUtil.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/p2/ui/operations/ProvisioningUtil.java
@@ -30,7 +30,6 @@ import org.eclipse.equinox.p2.metadata.repository.IMetadataRepository;
 import org.eclipse.equinox.p2.metadata.repository.IMetadataRepositoryManager;
 import org.eclipse.equinox.p2.ui.IProvisioningListener;
 import org.eclipse.equinox.p2.ui.ProvUIActivator;
-import org.eclipse.jface.util.PropertyChangeEvent;
 import org.eclipse.osgi.util.NLS;
 
 /**
@@ -74,7 +73,7 @@ public class ProvisioningUtil {
 		IMetadataRepository repo = manager.getRepository(location);
 		if (repo != null)
 			manager.removeRepository(repo);
-		PropertyChangeEvent event = new PropertyChangeEvent(repo, IProvisioningListener.REPO_REMOVED, null, null);
+		EventObject event = new EventObject(IProvisioningListener.REPO_REMOVED);
 		ProvUIActivator.getDefault().notifyListeners(event);
 
 	}
@@ -119,9 +118,8 @@ public class ProvisioningUtil {
 			IArtifactRepository repo = repos[i];
 			if (repo.getLocation().equals(location)) {
 				manager.removeRepository(repo);
-				PropertyChangeEvent event = new PropertyChangeEvent(repo, IProvisioningListener.REPO_REMOVED, null, null);
+				EventObject event = new EventObject(IProvisioningListener.REPO_REMOVED);
 				ProvUIActivator.getDefault().notifyListeners(event);
-
 				return;
 			}
 		}
